automations/python/label_pr.py

```diff
@@ -195,7 +195,11 @@ def main():
             log.info(f"Aspect label: {aspect}")
             log.info(f"Priority label: {priority}")
             log.info(f"Goal label: {goal}")
-            pr.set_labels(aspect, priority, goal)
+            labels_to_add = [aspect, priority, goal]
+            if stack := get_label_of_cat("stack", labels):
+                log.info(f"Stack label: {stack}")
+                labels_to_add.append(stack)
```

Comment:
This can be an issue because it will allow the labeller action to pass (only applying aspect, priority and goal labels) while causing the `check_stack_label` action to fail when the stack label is absent.

If that feels okay to do, I would recommend doing the same for the aspect, priority and goal labels so that the action applies as many labels as it can reliably infer from the issue. Currently it's all or none for these 3 categories.

Reply:
+1 to making the stack labels mandatory just like the other three. It would help us filter issues and PRs.

Reply:
I think "all or none" becomes a more difficult approach the more mandatory labels we have, since it adds even more work (which could have already been accomplished in an automated way) by maintainers when something goes wrong. If it's okay with y'all, I'll modify this approach so the PR labeling adds as many labels as it can, then we can rely on the label checks to catch the ones that are missing.

Reply:
I agree with setting as many labels as can be inferred (along with the awaiting triage) if not all labels could be inferred.

Reply:
Ahhh, that's a great call! I'll make that change as well.

Reply:
Done in 68f2292ab28217a1b1d3639ce8fe1dfb129897f6